IonDotnet.Tests/Internals/TreeReaderTest.cs

```diff
@@ -0,0 +1,251 @@
+        }
+
+        [TestMethod]
+        public void FlatStructScalarTest()
+        {
+            //Must be a flat struct of scalar values:
+            //boolean:true
+            //str:"yes"
+            //integer:123456
+            //longInt:int.Max*2
+            //bigInt:long.Max*10
+            //double:2213.1267567f
+            var value = new IonStruct
+            {
+                { "boolean", (IIonValue)_ionValueFactory.NewBool(true) },
+                { "str", (IIonValue)_ionValueFactory.NewString("yes") },
+                { "integer", (IIonValue)_ionValueFactory.NewInt(123456) },
+                { "longInt", (IIonValue)_ionValueFactory.NewInt((long)int.MaxValue * 2) },
+                { "bigInt", (IIonValue)_ionValueFactory.NewInt(BigInteger.Multiply(new BigInteger(long.MaxValue), 10)) },
+                { "double", (IIonValue)_ionValueFactory.NewFloat(2213.1267567) }
+            };
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.FlatScalar(reader);
+        }
+
+        [TestMethod]
+        public void SingleSymbolTest()
+        {
+            //{single_symbol:'something'}
+            var value = new IonStruct {{ "single_symbol", (IIonValue)_ionValueFactory.NewSymbol("something")}};
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.SingleSymbol(reader);
+        }
+
+        [TestMethod]
+        public void NestedAndCombinedListStructTest()
+        {
+            //Must be:
+            // {
+            //   menu: {  
+            //     id: "file",
+            //     popup: [    
+            //       "Open",
+            //       "Load",
+            //       "Close"
+            //     ],
+            //     deep1: {    
+            //       deep2: {      
+            //         deep3: {        
+            //           deep4val: "enddeep"
+            //         }
+            //       }
+            //     },
+            //     positions: [    
+            //       1234,
+            //       5678,
+            //       90
+            //     ]
+            //   }
+            // }
+
+            var popupList = (IIonValue)_ionValueFactory.NewEmptyList();
+            popupList.Add((IIonValue)_ionValueFactory.NewString("Open"));
+            popupList.Add((IIonValue)_ionValueFactory.NewString("Load"));
+            popupList.Add((IIonValue)_ionValueFactory.NewString("Close"));
+
+            var positionList = (IIonValue)_ionValueFactory.NewEmptyList();
+            positionList.Add((IIonValue)_ionValueFactory.NewInt(1234));
+            positionList.Add((IIonValue)_ionValueFactory.NewInt(5678));
+            positionList.Add((IIonValue)_ionValueFactory.NewInt(90));
+
+            var deep3 = new IonStruct {{ "deep4val", (IIonValue)_ionValueFactory.NewString("enddeep") }};
+
+            var deep2 = new IonStruct{{ "deep3", deep3 }};
+
+            var deep1 = new IonStruct { { "deep2", deep2 } };
```

Comment:
The spacing is inconsistent with the previous two values ( `{ {` vs `{{` ). Also, the previous value has no space between `IonStruct` and `{`, while the others do.